### ISSUE TITLE
Upgrade to 4.0-beta7 and locals data missing in template

### ISSUE THREAD

**Issue by bishopZ** [keystonejs/keystone#4646]
### Expected behavior
I expect the site to render.

### Actual/Current behavior
Am attempting to upgrade from 0.3.22 to 4.0-beta7 to resolve the security warning that github is giving about version 0.3.22.

After changing all the S3File fields to File fields. I received an error that the jade templates are not receiving any data. 

### Steps to reproduce the actual/current behavior

Model.js

```
Work.add(
	{
		metaDescription: { type: Types.Text },
		metaKeywords: { type: Types.Text },
		pageTitle: { type: Types.Text },
		heroImage: {
			type: Types.File,
			storage: s3Storage,
			note: 'Image dimensions should be approximately 1400x700'
		},
		slug: { type: String, required: true, initial: true, readonly: true },
		title: { type: Types.Text },
		subTitle: { type: Types.Text }
	},
//...etc
```

View.js
```
  view.on('init', function(next) {
    CaseStudyArchive.model.findOne()
      .populate('footer')
      .exec(function(err, result) {
        if(result) {
          locals.footer = result.footer;
          locals.workLanding = result;
          locals.title = result.pageTitle;
          locals.metaDescription = result.metaDescription;
          locals.metaKeywords = result.metaKeywords;
          locals.ogImage = result.heroImage.url;
          locals.ogUrl = req.path;
        } else {
          next();
        }
      });

//... etc

view.render('Template');
```
Template.jade
```
block content
  section#client_wrap.case-study-landing
    .detail_title_wrap(style="background: url(#{workLanding.heroImage.url}) top center / cover no-repeat fixed;")
```

The error I get is `can not read property heroImage of undefined`.

This code worked perfectly under 0.3, but with 4.0, none of the locals are getting into the template render. Is there some other thing I need to do when rendering a view

### Environment

| Software         | Version
| ---------------- | -------
| Keystone         |  4.0-beta7
| Node             |  6.11.1, 7.10.1, 8.10.0 

### Alternative Solution

I am only upgrading to the beta because of this message in Github

```
We found potential security vulnerabilities in your dependencies.

4 known vulnerabilities found
CVE-2017-15878 Moderate severity
CVE-2017-15879 Moderate severity
CVE-2017-15881 Moderate severity
CVE-2017-16570 High severity
package-lock.json update suggested:
keystone ~> 4.0.0-beta.7
```
Perhaps there is another solution?

**Comment by stennie** [comment]
@bishopZ Were you able to resolve this issue?

I suspect this may be due to `jade` no longer being included as an explicit dependency for Keystone 4 as noted in the [Upgrade Guide for 0.3 to 4.0](http://keystonejs.netlify.com/guides/v-0-3-to-v-4-0-upgrade-guide):

> Keystone 3 included `jade` as a dependency; this meant your project may have been able to use jade as the view engine without depending on it explicitly. Jade has now been removed from Keystone, so if you're using it you will need to explicitly add it to your `package.json`.

Regards,
Stennie

**Comment by bishopZ** [comment]
Thanks Stennie. I did notice the dependency change. If I had not set `jade` to be an explicit dependency, then I get the error `module jade can not be found`.

We have updated our dependencies to include `jade`.

I have noticed that we may have updated our version of Mongo at the same time as upgrading Keystone- changing its version from `"mongodb": "^2.1.16"` to `"mongodb": "^2.2.19",`. 

I will do some more tests to see if this might be the source of the error.
